### PR TITLE
chore: Angular 18 support

### PR DESCRIPTION
**Pull request by lukaVarga**

### CODE REVIEW

File: downstream_projects.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
   "example": "./example",
-  "sample-app-angular-hybrid": "https://github.com/ui-router/sample-app-angular-hybrid.git"
+  "sample-app-angular-hybrid": "https://github.com/lukaVarga/ui-router-sample-app-angular-hybrid.git"
```
Comment (wawyed) [inline downstream_projects.json:3]: revert back

Comment (lukaVarga) [reply downstream_projects.json]: @wawyed done

File: package.json [hunk 2]
```diff
@@ -28,14 +28,14 @@
...
+    "@angular/common": "^18.0.6",
+    "@angular/compiler": "^18.0.6",
+    "@angular/compiler-cli": "^18.0.6",
+    "@angular/core": "^18.0.6",
+    "@angular/platform-browser": "^18.0.6",
+    "@angular/platform-browser-dynamic": "^18.0.6",
+    "@angular/upgrade": "^18.0.6",
     "@types/angular": "^1.8.9",
     "@types/jquery": "^3.5.6",
     "@uirouter/angular": "^13.0.0",
```
Comment (wawyed) [inline package.json:41]: This needs to be 14

Comment (lukaVarga) [reply package.json]: Updated

Comment (wawyed) [reply package.json]: I think the examples folder needs updating to angular 18, etc,

Comment (lukaVarga) [reply package.json]: Right, I missed that one. I updated it now.

**Pull request merged**